analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -0,0 +1,533 @@
+```
+
+No, none of these identifiers are in the original putative oncogenic list.
+What about the unfiltered fusion files?
+
+STARFusion first.
+
+```{r}
+starfusion_df <- read_tsv(file.path(data_dir, "pbta-fusion-starfusion.tsv.gz"))
+```
+
+```{r}
+all(missing_bsids %in% starfusion_df$tumor_id)
+```
+
+```{r}
+arriba_df <- read_tsv(file.path(data_dir, "pbta-fusion-arriba.tsv.gz"))
+```
+
+```{r}
+all(missing_bsids %in% arriba_df$tumor_id)
+```
+
+So all samples that are "missing" from the putative oncogenic fusion file are present in the Arriba file, but some are not present in the STARFusion file.
+Putative oncogenic fusions are retained if they are detected by _either_ caller.
+We will consider biospecimen identifiers that were present in the Arriba file to have no evidence for these fusions, rather than consider them missing data.
+
+```{r}
+fusion_df <- fusion_df %>%
+  # Add in the instances where we have no evidence 
+  bind_rows(data.frame(Sample = missing_bsids)) %>%
+  replace(is.na(.), "None")
+```
+
+```{r}
+rm(arriba_df, putative_oncogenic_df, starfusion_df, missing_bsids)
+```
+
+### Join the Expression and Fusion Data
+
+```{r}
+rna_df <- left_join(exp_df, fusion_df,
+                    by = c("Kids_First_Biospecimen_ID" = "Sample"))
+head(rna_df, n = 10)
+```
+
+```{r}
+rna_df <- histologies_df %>%
+  select(Kids_First_Participant_ID,
+         sample_id, 
+         Kids_First_Biospecimen_ID) %>%
+  inner_join(rna_df)
+```
+
+```{r}
+rm(fusion_df, exp_df)
+```
+
+Are there duplicated `sample_id` in the RNA data frame?
+
+```{r}
+any(duplicated(rna_df$sample_id))
+```
+What are they?
+
+```{r}
+histologies_df %>%
+  filter(sample_id %in% rna_df$sample_id[which(duplicated(rna_df$sample_id))],
+         experimental_strategy == "RNA-Seq") %>%
+  group_by(sample_id) %>%
+  summarize(RNA_library = paste(RNA_library, collapse = ", "))
+```
+
+There are 2 instances where there are two stranded biospecimen IDs that map to the same `sample_id`.
+
+### All Data
+
+```{r}
+all_data_df <- left_join(dna_df, rna_df,
+                         by = c("Kids_First_Participant_ID",
+                                "sample_id")) %>%
+  rename(Kids_First_Biospecimen_ID_DNA = Kids_First_Biospecimen_ID.x,
+         Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID.y) %>%
+  select(starts_with("Kids_First"), sample_id, everything())
+
+head(all_data_df, n = 10)
+```
+
+Add in brain region and age.
+
+```{r}
+relevant_clinical_df <- histologies_df %>%
+  filter(sample_id %in% all_data_df$sample_id,
+         sample_type == "Tumor",
+         composition == "Solid Tissue") %>%
+  mutate(age_at_diagnosis_yr = 
+           floor(as.integer(age_at_diagnosis_days)/365)) %>%
+  group_by(Kids_First_Participant_ID, sample_id) %>%
+  summarize(glioma_brain_region = paste(sort(unique(glioma_brain_region)), 
+                                        collapse = ", "),
+            age_at_diagnosis_yr = paste(sort(unique(age_at_diagnosis_yr)),
```

Comment:
I would probably keep this handling for `glioma_brain_region` -- that seems like there's more opportunity for discrepancies but yea I could presumably randomly select an age values since it is age at diagnosis.

Reply:
No discrepancies, which means that it's functionally the same as randomly selecting a value (there are no trailing `,`) - I think I will leave it as is because any discrepancies would be obvious in the final table?